### PR TITLE
add to ac matching case for single elements/associations

### LINKED ISSUE #3344: Set simplification rule not applied (incomplete AC matching)
With simplification rules in this `test.k` module:
```
module TEST
    imports INT
    imports BOOL
    imports SET
    imports SET-SYMBOLIC
    imports K-EQUAL

    configuration <k> $PGM:Step </k>
    syntax Step ::= run ( Set ) | done ( Set )
    rule <k> run(S) => done(S) ... </k>

    rule [set-1]: (SetItem(X) REST) -Set SetItem(Y) => SetItem(X) (REST -Set SetItem(Y))
      requires X =/=Int Y [simplification]

    rule [set-2]: SetItem(X) -Set (SetItem(Y) REST) => SetItem(X) -Set REST
      requires X =/=Int Y [simplification]

    rule [set-3]: SetItem(X) -Set REST => SetItem(X)
      requires notBool (X in REST) [simplification]

endmodule
```
we would expect this claim in `test-spec.k` to be provable 
```
requires "test.k"
module TEST-SPEC
    imports TEST
    claim [needed]: <k> run( (SetItem(1) -Set (SetItem(2) REST)) REST) => done(SetItem(1) REST) ... </k>
        requires notBool (1 in REST)
endmodule
```
(i.e., the set is simplified as expected, using `set-2` and then `set-3` above)

Equation debugging shows that rule `set-2` is not applied because of an incomplete AC matching implementation.

### PR DESCRIPTION
**Pull request by jberthold**
Fixes #3344 

### Scope:
A single-element match is added, which was previously not happening because the cases in the AC matching code required an opaque rest.

Surprisingly, we had a unit test to check the prior behaviour, which was saying that subject `[0]` should _not_ match a pattern `[X] `  (obviously, `X = 0` is a matching substitution).

### Estimate:

---

###### Review checklist

The author performs the actions on the checklist. The reviewer evaluates the work and checks the boxes as they are completed.

-   [ ] **Summary.** Write a summary of the changes. Explain what you did to fix the issue, and why you did it. Present the changes in a logical order. Instead of writing a summary in the pull request, you may push a clean Git history.
-   [ ] **Documentation.** Write documentation for new functions. Update documentation for functions that changed, or complete documentation where it is missing.
-   [ ] **Tests.** Write unit tests for every change. Write the unit tests that were missing before the changes. Include any examples from the reported issue as integration tests.
-   [ ] **Clean up.** The changes are already clean. Clean up anything near the changes that you noticed while working. This does not mean only spatially near the changes, but logically near: any code that interacts with the changes!

### CODE REVIEW

File: kore/test/Test/Kore/Rewrite/Axiom/Matcher.hs [hunk 1]
```diff
@@ -1078,6 +1079,13 @@ test_matching_Set =
         (mkVar sSet)
         (mkSet [mkElemVar xInt] [])
         [(sSet, mkSet [mkElemVar xInt] [])]
+    , matches
+        "[x:Int] s:Set matches [1] s':Set"
+        (mkSet [mkElemVar xInt] [mkVar sSet])
+        (mkSet [mkInt 1] [mkVar s'Set])
+        [ (inject xInt, mkInt 1)
+        , (sSet, mkSet [] [mkVar s'Set])
+        ]
```
Comment (jberthold) [inline kore/test/Test/Kore/Rewrite/Axiom/Matcher.hs:1088]: This was actually passing before, I thought the bug was here before I traced the execution

File: test/issue-3344/test-spec.k [hunk 2]
```diff
@@ -0,0 +1,9 @@
+requires "test.k"
+
+module TEST-SPEC
+    imports TEST
+
+    claim [needed]: <k> run( (SetItem(1) -Set (SetItem(2) REST)) REST) => done(SetItem(1) REST) ... </k>
+        requires notBool (1 in REST)
+
```
Comment (ehildenb) [inline test/issue-3344/test-spec.k:8]: Can we also add the claim for `requires 1 in REST` too? I believe that was already passing before anyway, but good to codify it as law.

Comment (jberthold) [reply test/issue-3344/test-spec.k:8]: Have added the second test.
Unfortunately this integration test has to wait for an update to the K version dependency.
```
env -u KORE_EXEC_OPTS kompile --backend haskell --haskell-backend-command kore-exec -d . test.k
Error:  Compiler: Could not find module: SET-SYMBOLIC
        Source(/build/kore/test/issue-3344/test.k)
        Location(5,5,5,25)
        5 |         imports SET-SYMBOLIC
          .         ^~~~~~~~~~~~~~~~~~~~
make[1]: *** [/build/kore/test/issue-3344/../include.mk:51: test-kompiled/definition.kore] Error 113
```

**Comment by jberthold** [comment]
Blocked on #3343 to make the new integration tests work.

**Review by ana-pantilie — APPROVED** [review]

**Pull request merged**